### DISCUSSION TITLE
SvelteKit + TailwindCSS

### DISCUSSION THREAD

**Discussion by tohodakilla** [tailwindlabs/tailwindcss#7984] [Help] [answered]
Hello!

I have installed tailwind with svelteKit according to latest instructions here: https://tailwindcss.com/docs/guides/sveltekit
I'm running node v16.14.2, tailwind 3.0.23, svelteKit 303

Everything works fine untill I try to generate tailwind classes using svelteKit functions.

`<button class="px-6 py-2 rounded border-green-500 hover:bg-green-500 text-slate-100">BUTTON</button>`
above works just fine, but when I try something like this:

```
<script>
const colors = ['gray', 'zinc', 'neutral', 'red', 'orange', 'amber', 'yellow']
</script>

{#each colors as color}
<button class="px-6 py-2 rounded border-{color}-500 hover:bg-green-{color} text-slate-{color}">BUTTON</button>
{/each}
```
it does not work, no tailwind css for those generated classes is present. only when i copy result from source and paste to apropriate svelte file it all generates correctly and works fine until i re-run "npm run dev"

please point me in right direction i've tried to find information about this issue but failed.

**Comment by NovusOpus** [comment, marked as answer]
This is not working because you're trying to create dynamic classes, and since Tailwind doesn’t actually run your code, it won’t detect dynamically constructed class names.

This is explained in the docs: [Dynamic class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names)
